cli/src/cli_util.rs

```diff
@@ -1012,7 +1012,22 @@ impl WorkspaceCommandHelper {
         let immutable = revset_util::parse_immutable_expression(&self.revset_parse_context())?;
         let mut expression = self.attach_revset_evaluator(immutable)?;
         expression.intersect_with(&to_rewrite_revset);
-        if let Some(commit_id) = expression.evaluate_to_commit_ids()?.next() {
+        if let Some(commit_id) = expression
+            .evaluate_to_commit_ids()
+            .map_err(|e| {
+                let UserRevsetEvaluationError::Resolution(RevsetResolutionError::NoSuchRevision {
+                    name,
+                    candidates: _,
+                }) = e
+                else {
+                    return e.into();
+                };
+                user_error_with_hint(format!(
+                    r#"The revision "{name}" used in immutable_heads() does not exist"#
+                ), r#"If you configured specific branch names consider wrapping them with "present()" or using the functions "branches()" / "remote_branches()"."#)
+            })?
```

Comment:
Do you think it's time to extract the iterator to a variable?

```rust
let commit_id_iter = ...;
if let Some(commit_id) = commit_id_iter.next() {
  ...
}

```

Reply:
I think that's more readable yeah